server/api/unsafe_operation.go

```diff
@@ -36,22 +36,43 @@ func newUnsafeOperationHandler(svr *server.Server, rd *render.Render) *unsafeOpe
 
 // @Tags unsafe
 // @Summary Remove failed stores unsafely.
+// @Accept json
+// @Param body body object true "json params"
 // @Produce json
 // Success 200 {string} string "Request has been accepted."
 // Failure 400 {string} string "The input is invalid."
 // Failure 500 {string} string "PD server failed to proceed the request."
 // @Router /admin/unsafe/remove-failed-stores [POST]
 func (h *unsafeOperationHandler) RemoveFailedStores(w http.ResponseWriter, r *http.Request) {
 	rc := getCluster(r)
-	var stores map[uint64]string
-	if err := apiutil.ReadJSONRespondError(h.rd, w, r.Body, &stores); err != nil {
+	var input map[string]interface{}
+	if err := apiutil.ReadJSONRespondError(h.rd, w, r.Body, &input); err != nil {
+		h.rd.JSON(w, http.StatusBadRequest, "Cannot parse the request")
 		return
 	}
-	if len(stores) == 0 {
+	rawStores, exists := input["stores"]
+	if !exists {
 		h.rd.JSON(w, http.StatusBadRequest, "No store specified")
 		return
 	}
-	if err := rc.GetUnsafeRecoveryController().RemoveFailedStores(stores); err != nil {
+	storeList := rawStores.([]interface{})
+	if len(storeList) == 0 {
+		h.rd.JSON(w, http.StatusBadRequest, "Invalid input for the stores")
+		return
+	}
+	stores := make(map[uint64]string)
+	for _, rawStore := range storeList {
+		stores[uint64(rawStore.(float64))] = ""
+	}
+	var timeout int64
```

Comment:
```suggestion
	timeout := int64(600)
	rawTimeout, exists := input["timeout"].(float64)
	if exist {
	    timeout = int64(rawTimeout)
	}
```

Reply:
Changed, slightly different, since we can't put `.(float64)` right after extracting the value from the map.